storm-core/src/jvm/org/apache/storm/daemon/StormCommon.java

```diff
@@ -436,7 +414,9 @@ public static void addMetricComponents(Map conf, StormTopology topology) {
     public static void addSystemComponents(Map conf, StormTopology topology) {
         Map<String, StreamInfo> outputStreams = new HashMap<String, StreamInfo>();
         outputStreams.put(Constants.SYSTEM_TICK_STREAM_ID, Thrift.outputFields(Arrays.asList("rate_secs")));
+        outputStreams.put(Constants.METRICS_STREAM_ID, Thrift.directOutputFields(Arrays.asList("task-info", "data-points")));
```

Comment:
Yes we send the metrics tick tuple to each task with source stream to that, so it's needed.